lib/engine/auto_router.rb

```diff
@@ -88,81 +133,328 @@ def compute(corporation, **opts)
+    #   hexside_bits is a map of hexside_id to bit number
+    #   bit_to_bitmask is a helper table of bit 0-31 to integer bitmask (essentially 1 << bit)
+    # returns:
+    #   the bitfield (array of ints) representing all hexsides in the connection path
+    # updates:
+    #   new hexsides are added to hexside_bits
+    def bitfield_from_connection(connection, hexside_bits, bit_to_bitmask)
+      bitfield = [0]
+      connection.each do |conn|
+        paths = conn[:chain][:paths]
+        index = 1
+        stop = paths.size
+        node2 = paths[0]
+        while index < stop
+          # hand-optimized ruby gives faster opal code
+          node1 = node2
+          node2 = paths[index]
+          case node1.edges.size
+          when 1
+            # node1 has 1 edge, connect it to first edge of node2
+            hexside_left = node1.edges[0].id
+            hexside_right = node2.edges[0].id
+            check_and_set(bitfield, hexside_left, hexside_right, hexside_bits, bit_to_bitmask)
+          when 2
+            # node1 has 2 edges, connect them as well as 2nd edge to first node2 edge
+            hexside_left = node1.edges[0].id
+            hexside_right = node1.edges[1].id
+            check_and_set(bitfield, hexside_left, hexside_right, hexside_bits, bit_to_bitmask)
+            hexside_left = hexside_right
+            hexside_right  = node2.edges[0].id
+            check_and_set(bitfield, hexside_left, hexside_right, hexside_bits, bit_to_bitmask)
+          else
+            puts "  ERROR: auto-router found unexpected number of path node edges #{node1.edges.size}. "\
+                 'Route combos may be be incorrect'
           end
+          index += 1
         end
+      end
+      bitfield
+    end
+
+    def check_and_set(bitfield, hexside_left, hexside_right, hexside_bits, bit_to_bitmask)
+      check_edge_and_set(bitfield, hexside_left, hexside_bits, bit_to_bitmask)
+      check_edge_and_set(bitfield, hexside_right, hexside_bits, bit_to_bitmask)
+    end
 
-        combos.compact!
-      rescue RuntimeError
-        puts 'Route timeout reach'
-        break
+    def check_edge_and_set(bitfield, hexside_edge, hexside_bits, bit_to_bitmask)
+      if hexside_bits.include?(hexside_edge)
+        set_bit(bitfield, hexside_bits[hexside_edge], bit_to_bitmask)
+      else
+        newbit = hexside_bits.length.positive? ? hexside_bits.map { |_hexside, bit| bit }.max + 1 : 0
+        hexside_bits[hexside_edge] = newbit
+        set_bit(bitfield, newbit, bit_to_bitmask)
       end
+    end
 
-      puts "Found #{possibilities.size} possible routes in: #{Time.now - now}"
+    # bitfield is an array of integers, can be expanded by this call if necessary
+    # bit is a bit number, 0 is lowest bit, 32 will jump to the next int in the array, and so on
+    def set_bit(bitfield, bit, bit_to_bitmask)
+      entry = (bit / 32).to_i
+      mask = bit_to_bitmask[bit & 31]
+      add_count = entry + 1 - bitfield.size
```

Comment:
this code looks convoluted, are you trying to to do n.times.each { bitfield << 0 } 

or bitfield.concat([0] * n)